src/todoitem_service.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC, abstractmethod
+from src.schemas import TODOItem, UpdateTODOItem, BaseTODOItem
+from fastapi import Response, status, HTTPException
+
+
+class ITODOItemsService(ABC):
+
+    @abstractmethod
+    def get_all_todoitems(self) -> list[TODOItem]:
+        """Returns a list of all todoitems"""
+
+    @abstractmethod
+    def create_todoitem(self, todoitem: BaseTODOItem) -> TODOItem:
+        """Creates new todoitem"""
+
+    @abstractmethod
+    def get_todoitem_by_id(self, id: int) -> TODOItem:
+        """Creates new todoitem"""
+
+    @abstractmethod
+    def update_todoitem(self, id: int, new_data: UpdateTODOItem) -> TODOItem:
+        """Updates a todoitem"""
+
+    @abstractmethod
+    def remove_todoitem(self, id: int) -> None:
```

Comment:
This says you return nothing, but in the implementation you are returning a dict?